integration-tests/src/tests/client/features/increase_storage_compute_cost.rs

```diff
@@ -0,0 +1,369 @@
+            Path::new("."),
+            create_test_store(),
+            &genesis,
+            TrackedConfig::new_empty(),
+            runtime_config_store,
+        );
+        TestEnv::builder(chain_genesis).runtime_adapters(vec![runtime]).build()
+    };
+
+    // setup: deploy the contract
+    {
+        // This contract has a bunch of methods to invoke storage operations.
+        let code = near_test_contracts::rs_contract().to_vec();
+        let actions = vec![Action::DeployContract(DeployContractAction { code })];
+
+        let signer = InMemorySigner::from_seed(
+            contract_account.clone(),
+            KeyType::ED25519,
+            &contract_account,
+        );
+        let tx = env.tx_from_actions(actions.clone(), &signer, signer.account_id.clone());
+        env.execute_tx(tx).unwrap().assert_success();
+    }
+
+    // Fetching the correct nonce from `env` is a bit fiddly, we would have to
+    // query the access key of the user. It's easier to keep a shared counter
+    // that starts at 1 and increases monotonically.
+    let mut nonce = 1;
+
+    let old_chunk = produce_saturated_chunk(
+        &mut env,
+        &user_account,
+        &contract_account,
+        method_name.clone(),
+        method_args.clone(),
+        num_transactions,
+        should_fail,
+        old_config.as_ref(),
+        &mut nonce,
+    );
+    let chunk_header = old_chunk.cloned_header();
+    let gas_burnt = chunk_header.gas_used();
+    let gas_limit: u64 = chunk_header.gas_limit();
+    assert!(
+        gas_burnt >= gas_limit,
+        "should saturate gas limit, only burnt {gas_burnt} when limit was {gas_limit}"
+    );
+
+    env.upgrade_protocol(new_protocol_version);
+
+    let new_chunk = produce_saturated_chunk(
+        &mut env,
+        &user_account,
+        &contract_account,
+        method_name,
+        method_args,
+        num_transactions,
+        should_fail,
+        new_config.as_ref(),
+        &mut nonce,
+    );
+
+    let old_receipts_num = old_chunk.receipts().len();
+    let new_receipts_num = new_chunk.receipts().len();
+    if uses_storage {
+        assert!(
+            new_receipts_num < old_receipts_num,
+            "should reach compute limit before gas limit (receipts before: {} receipts now: {})",
+            old_receipts_num,
+            new_receipts_num,
+        );
+    } else {
+        assert_eq!(old_receipts_num, new_receipts_num, "compute costs should not affect this test");
+    }
+}
+
+/// Saturate a chunk with function call receipts and returns that chunk.
+///
+/// (This is a helper function called twice by the helper function above, once
+/// before and once after the upgrade.)
+///
+/// This function creates many function call receipts with the give signer,
+/// receiver, method name, and method argument. Then it submits it to a client
+/// and produces a few blocks. Then it returns the `gas_used` value of the first
```

Comment:
I couldn't find where `gas_used` is returned. Is this part of comment outdated?

Reply:
Oh yes, this is outdated. I just return the entire chunk now. Will update the comment.